# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [4]:
import pandas as pd
import numpy as np

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [5]:
s1 = pd.Series(np.random.randint(low=1, high=5, size=100))
s2 = pd.Series(np.random.randint(1, high=4, size=100))
s3 = pd.Series(np.random.randint(10000, 30001, 100))

print(s1, s2, s3)

0     3
1     1
2     2
3     2
4     1
     ..
95    4
96    4
97    3
98    3
99    2
Length: 100, dtype: int32 0     1
1     3
2     1
3     3
4     3
     ..
95    3
96    1
97    2
98    3
99    1
Length: 100, dtype: int32 0     11556
1     24453
2     14455
3     24979
4     29995
      ...  
95    22726
96    25898
97    19915
98    20176
99    29487
Length: 100, dtype: int32


### Step 3. Let's create a DataFrame by joinning the Series by column

In [6]:
housemkt = pd.concat([s1, s2, s3], axis=1)
housemkt.head()

0  1      2
0  3  1  11556
1  1  3  24453
2  2  1  14455
3  2  3  24979
4  1  3  29995

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [7]:
housemkt.columns = ['bedrs', 'bathrs', 'price_sqr_meter']
housemkt

bedrs  bathrs  price_sqr_meter
0       3       1            11556
1       1       3            24453
2       2       1            14455
3       2       3            24979
4       1       3            29995
..    ...     ...              ...
95      4       3            22726
96      4       1            25898
97      3       2            19915
98      3       3            20176
99      2       1            29487

[100 rows x 3 columns]

In [8]:
housemkt.rename(columns={0: 'bedrs', 1: 'bathrs', 2: 'price_sqr_meter'}, inplace=True)
housemkt.head()

bedrs  bathrs  price_sqr_meter
0      3       1            11556
1      1       3            24453
2      2       1            14455
3      2       3            24979
4      1       3            29995

### Step 5. Create a column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [9]:
bigcolumn2 = pd.concat([s1, s2, s3], axis=0)
bigcolumn2

0         3
1         1
2         2
3         2
4         1
      ...  
95    22726
96    25898
97    19915
98    20176
99    29487
Length: 300, dtype: int32

In [10]:
one_column = pd.concat((s1, s2, s3), ignore_index=True)
one_column

0          3
1          1
2          2
3          2
4          1
       ...  
295    22726
296    25898
297    19915
298    20176
299    29487
Length: 300, dtype: int32

In [11]:
bigcolumn0 = pd.concat([s1, s2, s3], axis= 0)
bigcolumn4 = pd.DataFrame(bigcolumn0)
bigcolumn4

0
0       3
1       1
2       2
3       2
4       1
..    ...
95  22726
96  25898
97  19915
98  20176
99  29487

[300 rows x 1 columns]

In [12]:
# join concat the values
bigcolumn = pd.concat([s1, s2, s3], axis=0)

# it is still a Series, so we need to transform it to a DataFrame
bigcolumn = bigcolumn.to_frame()
print(type(bigcolumn))

bigcolumn

<class 'pandas.core.frame.DataFrame'>


0
0       3
1       1
2       2
3       2
4       1
..    ...
95  22726
96  25898
97  19915
98  20176
99  29487

[300 rows x 1 columns]

### Step 6. Oops, it seems it is going only until index 99. Is it true?

In [13]:
bigcolumn.index.values

array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33,
       34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50,
       51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67,
       68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84,
       85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99,  0,  1,
        2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16, 17, 18,
       19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35,
       36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52,
       53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69,
       70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86,
       87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99,  0,  1,  2,  3,
        4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20,
       21, 22, 23, 24, 25

In [14]:
# no the index are kept but the length of the DataFrame is 300
len(bigcolumn)

300

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [15]:
help(bigcolumn.reset_index)

Help on method reset_index in module pandas.core.frame:

reset_index(level: Union[Hashable, Sequence[Hashable], NoneType] = None, drop: bool = False, inplace: bool = False, col_level: Hashable = 0, col_fill: Union[Hashable, NoneType] = '') -> Union[ForwardRef('DataFrame'), NoneType] method of pandas.core.frame.DataFrame instance
    Reset the index, or a level of it.
    
    Reset the index of the DataFrame, and use the default one instead.
    If the DataFrame has a MultiIndex, this method can remove one or more
    levels.
    
    Parameters
    ----------
    level : int, str, tuple, or list, default None
        Only remove the given levels from the index. Removes all levels by
        default.
    drop : bool, default False
        Do not try to insert index into dataframe columns. This resets
        the index to the default integer index.
    inplace : bool, default False
        Modify the DataFrame in place (do not create a new object).
    col_level : int or str, default 0


In [16]:
bigcolumn.reset_index(drop=True, inplace=True)

In [17]:
bigcolumn

0
0        3
1        1
2        2
3        2
4        1
..     ...
295  22726
296  25898
297  19915
298  20176
299  29487

[300 rows x 1 columns]

In [18]:
bigcolumn.reset_index(drop=True, inplace=True)
bigcolumn2 = bigcolumn.loc[:, 0]
print(type(bigcolumn2))
bigcolumn2

<class 'pandas.core.series.Series'>


0          3
1          1
2          2
3          2
4          1
       ...  
295    22726
296    25898
297    19915
298    20176
299    29487
Name: 0, Length: 300, dtype: int32

In [19]:
bigcolumn.index.values

array([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,
        13,  14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,
        26,  27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,
        39,  40,  41,  42,  43,  44,  45,  46,  47,  48,  49,  50,  51,
        52,  53,  54,  55,  56,  57,  58,  59,  60,  61,  62,  63,  64,
        65,  66,  67,  68,  69,  70,  71,  72,  73,  74,  75,  76,  77,
        78,  79,  80,  81,  82,  83,  84,  85,  86,  87,  88,  89,  90,
        91,  92,  93,  94,  95,  96,  97,  98,  99, 100, 101, 102, 103,
       104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116,
       117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129,
       130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142,
       143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155,
       156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168,
       169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 18